### DISCUSSION TITLE
binomial logistic regression response variables

### DISCUSSION THREAD

**Discussion by Shan-square** [SchlossLab/mikropml#341] [Q&A] [answered]
Hello, Is there a way to code the response variable when it is a two-column matrix of counts when using binomial logistic regression with the following code:

run_ml(data, method="glmnet", outcome_colname="y", kfold=5, cv_times=100, training_frac=0.8)

I might be wrong, but it seems the outcome_colname option only allows one column as the input. I would like to use a "y" which includes the counts of individuals with disease and without disease. An example of "y" would be:

disease  no_disease
2             3
5             8
3             9
....

Thanks.

**Comment by kelly-sovacool** [comment]
Hi @Shan-square, mikropml and caret require a data frame where the `outcome_colname` is exactly one column, and all remaining columns are used as features (a.k.a predictor variables), with each row as an individual/observation. What are the features / predictors you wish to train the model on?

**Reply by Shan-square** [reply]
Thanks Kelly, I am using soil bacterial relative abundance data to predict the incidence of plant fruit disease. For each observation (row), the response variable contains the number of fruits with disease and the number of fruits without. I would like to keep both numbers for each observation because they are from the same plant. I have about 500 observations. If only one column is allowed, I can transform them into fractions and using linear regression. Do you have other suggestions for transforming/coding the response variable if I want to keep the binomial logistic regression?

**Reply by kelly-sovacool** [reply]
Binomial logistic regression is for outcomes/responses that are dichotomous, e.g. yes/no, pass/fail, whether _any_ fruits from a plant were diseased -- in which case you would still have only one outcome column. I would recommend transforming your response variable to fractions and using ~logistic~ (correction: linear) regression, e.g.
``` r
library(dplyr)
set.seed(20231208)
# create example data frame with dummy otu abundances
dat <- data.frame(disease=c(2,5,3), 
                  no_disease=c(3,8,9),
                  otu1 = runif(3),
                  otu2 = runif(3),
                  otu3 = runif(3))
dat
#>   disease no_disease       otu1      otu2       otu3
#> 1       2          3 0.77708972 0.8929019 0.74143156
#> 2       5          8 0.03412326 0.6826952 0.08643496
#> 3       3          9 0.12043893 0.1737634 0.78740345

# calculate fraction of fruits that were diseased for each plant
dat_frac <- dat %>% 
  mutate(frac_diseased = disease / (disease + no_disease), .before = otu1) %>%
  select(-c(disease, no_disease))
dat_frac
#>   frac_diseased       otu1      otu2       otu3
#> 1     0.4000000 0.77708972 0.8929019 0.74143156
#> 2     0.3846154 0.03412326 0.6826952 0.08643496
#> 3     0.2500000 0.12043893 0.1737634 0.78740345
```

<sup>Created on 2023-12-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by Shan-square** [reply]
Thanks Kelly! Do I need to do add anything to 
run_ml (data, method="glmnet", outcome_colname="frac_disease", kfold=5, cv_times=100, training_frac=0.8) 
to show that it is a logistic regression?

**Reply by kelly-sovacool** [reply]
Nope, that will work as-is as long as `data` is your reformatted dataframe. `glmnet` works generally for regression and will detect the type based on the outcome column. Do take a look at our vignettes if you haven't yet. https://www.schlosslab.org/mikropml/articles/introduction.html#the-methods-we-support

**Reply by kelly-sovacool** [reply, marked as answer]
Oh by the way, this will perform linear regression since the outcome column will be a continuous variable. Logistic regression is for categorical variables.

**Reply by Shan-square** [reply]
That sounds good. Thanks for your help Kelly!
